src/linux/sock_ctrl_msg.rs

```diff
@@ -220,28 +220,42 @@ fn raw_recvmsg(fd: RawFd, iovecs: &mut [iovec], in_fds: &mut [RawFd]) -> Result<
     }
 
     let mut cmsg_ptr = msg.msg_control as *mut cmsghdr;
-    let mut in_fds_count = 0;
-    while !cmsg_ptr.is_null() {
+    let mut copied_fds_count = 0;
+    let mut is_full = false;
+
+    while !cmsg_ptr.is_null() && !is_full {
         // Safe because we checked that cmsg_ptr was non-null, and the loop is constructed such that
-        // that only happens when there is at least sizeof(cmsghdr) space after the pointer to read.
+        // that it only happens when there is at least sizeof(cmsghdr) space after the pointer to
+        // read.
         let cmsg = unsafe { (cmsg_ptr as *mut cmsghdr).read_unaligned() };
 
         if cmsg.cmsg_level == SOL_SOCKET && cmsg.cmsg_type == SCM_RIGHTS {
             let fd_count = ((cmsg.cmsg_len - CMSG_LEN!(0)) as usize) / size_of::<RawFd>();
+            let mut end_idx = in_fds.len();
+
+            if copied_fds_count + fd_count >= end_idx {
+                is_full = true;
```

Comment:
When is_full becomes true, should we close all un-copied RawFds? Otherwise those un-copied fds will get leaked.

Reply:
I think we do not need to do this due to the fact that `cmsg_buffer` has the same capacity as `in_fds`. This means that when we receive ancillary data, we can hold at maximum `cmsg_buffer` capacity bytes, which corresponds as well to `in_fds` capacity, so copying from one to the other does not imply handling extra data. If the sender transmits more data than the receiver can hold, some data gets discarded. I've experimented with this in the added unit test. Please let me know if this makes sense.

Reply:
I'm hesitating about what to return in case of receiving unexpected data/fds. It's a sign of incorrect communication protocol, so it's a little risky to silently drop the unexpected data/fds. How about returning an error here?